src/Storages/MergeTree/MergeTreeIndexGranularityAdaptive.cpp

```diff
@@ -60,6 +66,8 @@ void MergeTreeIndexGranularityAdaptive::appendMark(size_t rows_count)
     {
         marks_rows_partial_sums.push_back(marks_rows_partial_sums.back() + rows_count);
     }
+
+    CurrentMetrics::add(CurrentMetrics::MergeTreeIndexGranularityInternalArraysTotalSize, sizeof(rows_count));
```

Comment:
should we also subtract it when the part gets removed?

Reply:
fixed